src/ServiceControl.Monitoring/Http/Diagrams/DiagramApiModule.cs

```diff
@@ -17,7 +17,7 @@ public class MonitoredEndpointsModule : ApiModule
         /// Initializes the metric API module.
         /// </summary>
         // ReSharper disable SuggestBaseTypeForParameter
-        public MonitoredEndpointsModule(IProvideBreakdownBy<EndpointInstanceId>[] metricByInstance, IProvideBreakdownBy<EndpointMessageType>[] metricByMessageType, EndpointRegistry endpointRegistry, EndpointInstanceActivityTracker activityTracker, MessageTypeRegistry messageTypeRegistry)
+        public MonitoredEndpointsModule(IProvideBreakdownBy<EndpointInstanceId>[] metricByInstance, IProvideBreakdownBy<EndpointMessageType>[] metricByMessageType, EndpointRegistry endpointRegistry, EndpointInstanceActivityTracker activityTracker, MessageTypeRegistry messageTypeRegistry, MonitoredEndpointMessageTypeParser messageTypeParser)
```

Comment:
I see the injection point but fail to find registration point. Does it meant that `MonitoredEndpointMessageTypeParser` will be a singleton. If not, the caching that is instance based won't work.

Reply:
The registration is done by the containers. All types are [singletons](https://github.com/Particular/ServiceControl.Monitoring/blob/develop/src/ServiceControl.Monitoring/EndpointFactory.cs#L150)

Reply:
Lovely!